scripts/lsh.py

```diff
@@ -309,6 +321,68 @@ def other_main(args):
 
 
+def collect_previous_dirs(path, deadline_date):
+    """ Collects the directories which are directly under the path given
+     and whose name, when interpreted as a date, are earlier than the
+     deadline_date
+
+     TODO If we want to use this functionality in other scripts, it should be
+      moved to utils.py."""
+
+    logging.info(f"We are looking for dirs older than {deadline_date} in {path}")
+    collected_dirs= []
+    deadline_year, deadline_month = deadline_date.split('_')
+    dirs_in_path = os.listdir(path)
+    for filename in dirs_in_path:
+        # We suppose that the directories obey our strict naming convention:
+        # y_m and that all years are written in the same number of digits.
+        if re.match('^[0-9]+_[0-9]+$', filename):
+            filename_year, filename_month = filename.split('_')
+            if filename_year > deadline_year:
+                continue
+            elif filename_year == deadline_year:
+                if filename_month >= deadline_month:
+                    continue
+            else:
+                collected_dirs.append(os.path.join(path, filename))
+    logging.info("The following directories have been collected as the cumulative past:")
+    logging.info(collected_dirs)
+    return collected_dirs
+
+def cumulative_directory_deduplication(args):
+    """The "real" main function of the "cumulative" mode. """
+
+    # We suppose here that the final part of the input directory is a
+    # date-like string e.g.: 06_filtered/2022_12/
+    input_date = os.path.basename(os.path.normpath(args.input_dir))
+    past_batches = collect_previous_dirs(args.cumulative_dir, input_date)
+    number_of_past_batches = len(past_batches)
+
+    with TemporaryDirectory() as tmp_root_dir:
+        current_input_dir = args.input_dir
+        i = 0
+        for past_batch in past_batches:
+            i += 1
+            if i == number_of_past_batches:
+                # This is the last cross-deduplication, the results will go to the
+                # final output dir
+                current_output_dir = args.output_dir
+            else:
+                past_batch_date = os.path.basename(past_batch)
+                # There are still cross-deduplications to do, the results will go
+                # to the tmp output dir.
+                current_output_dir = f"{tmp_root_dir}/{input_date}" \
+                                     f"_against_{past_batch_date}/"
+            logging.info(f"Cross-deduplicating {current_input_dir} with {past_batch}, "
+                  f"moving results to {current_output_dir}")
+            pairwise_directory_deduplication(current_input_dir, current_output_dir,
+                                             past_batch, args.processes,
+                                             args.permutations, args.threshold)
+            current_input_dir = current_output_dir
+    # TODO do we need any extra statistics to log at the end?
```

Comment:
Probably not, but you'll need to delete the temporary directory. Also, three empty lines are one too much (again, pyflakes).